src/navigation/cameraStackNavigation.js

```diff
@@ -2,28 +2,46 @@
 
 import * as React from "react";
 import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import { PermissionsAndroid } from "react-native";
 
-import PhotoGallery from "../components/PhotoLibrary/PhotoGallery";
 import GroupPhotos from "../components/PhotoLibrary/GroupPhotos";
 import ObsEdit from "../components/ObsEdit/ObsEdit";
-import PhotoGalleryProvider from "../providers/PhotoGalleryProvider";
 import SoundRecorder from "../components/SoundRecorder/SoundRecorder";
 import NormalCamera from "../components/Camera/NormalCamera";
 import CVSuggestions from "../components/ObsEdit/CVSuggestions";
 import CustomHeaderWithTranslation from "../components/SharedComponents/CustomHeaderWithTranslation";
+import PhotoGalleryProvider from "../providers/PhotoGalleryProvider";
+import PhotoGallery from "../components/PhotoLibrary/PhotoGallery";
+import PermissionGate from "../components/SharedComponents/PermissionGate";
 
 const Stack = createNativeStackNavigator( );
 
 const hideHeader = {
   headerShown: false
 };
 
+const PhotoGalleryWithPermission = ( ) => (
```

Comment:
What's your thinking behind using `PermissionGate` for the camera and photo gallery, but not the sound recorder? Wouldn't it be easiest to keep all Android permissions code in the same location?

Reply:
My thinking was just that I was trying to address the two issues I filed, but yes, if you're cool with this approach, I think we should apply it elsewhere after merging.

Reply:
Gotcha, sounds like a plan!